### PR TITLE
Add WebGPU rendering sample

### PR DESCRIPTION
**Pull request by sandersdan**
Additionally this PR refactors the sample to make it easy to swap renderers, adds status messages, and removes Origin Trial elements.

### CODE REVIEW

File: samples/video-decode-display/demuxer_mp4.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
+importScripts("../third_party/mp4boxjs/mp4box.all.min.js");
```
Comment (chcunningham) [inline samples/video-decode-display/demuxer_mp4.js:1]: Super clean!

File: samples/video-decode-display/index.html [hunk 2]
```diff
@@ -1,30 +1,77 @@
...
-  demuxDecodeWorker = new Worker("./demux_decode_worker.js");
-  demuxDecodeWorker.postMessage({ canvas: offscreen}, [offscreen]);
-</script>
+  <canvas></canvas>
+
+  <script type="module">
+    // Play button.
+    const startButton = document.querySelector("#start");
+    startButton.addEventListener("click", () => {
+      startButton.disabled = true;
```
Comment (chcunningham) [inline samples/video-decode-display/index.html:45]: can you disable the radio buttons too?

File: samples/video-decode-display/index.html [hunk 3]
```diff
@@ -1,30 +1,77 @@
...
+    const status = {
+      fetch: document.querySelector("#fetch"),
+      demux: document.querySelector("#demux"),
+      decode: document.querySelector("#decode"),
+      render: document.querySelector("#render"),
+    };
 
+    function setStatus(message) {
+      for (const key in message.data) {
+        if (status.hasOwnProperty(key)) {
```
Comment (chcunningham) [inline samples/video-decode-display/index.html:59]: nit: assert status.hasOwnProperty(key) ?

File: samples/video-decode-display/index.html [hunk 4]
```diff
@@ -1,30 +1,77 @@
...
+      <input id="renderer_2d" type="radio" name="renderer" value="2d"> 2D
+    </label>
+    <label for="renderer_webgl">
+      <input id="renderer_webgl" type="radio" name="renderer" value="webgl" checked> WebGL
+    </label>
+    <label for="renderer_webgl2">
+      <input id="renderer_webgl2" type="radio" name="renderer" value="webgl2"> WebGL 2
+    </label>
+    <label for="renderer_webgpu">
+      <input id="renderer_webgpu" type="radio" name="renderer" value="webgpu"> WebGPU
```
Comment (chcunningham) [inline samples/video-decode-display/index.html:24]: I'd put some disclaimer note that this requires a chrome flag or similar (and may generally be broken) while WebGPU is still being developed.

**Review by chcunningham — APPROVED** [review]
Awesome! Love the demuxer. Love the cleanup. Just a few nits.

File: samples/video-decode-display/demuxer_mp4.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    // Configure an MP4Box File for demuxing.
+    this.#file = MP4Box.createFile();
+    this.#file.onError = error => setStatus("demux", error);
+    this.#file.onReady = this.#onReady.bind(this);
+    this.#file.onSamples = this.#onSamples.bind(this);
+
+    // Fetch the file and pipe the data through.
+    const fileSink = new MP4FileSink(this.#file, setStatus);
+    fetch(uri).then(response => {
+      response.body.pipeTo(new WritableStream(fileSink, {highWaterMark: 2}));
```
[the remark this answers is not in the document]
Comment (sandersdan) [reply samples/video-decode-display/demuxer_mp4.js:58]: Done.

File: samples/video-decode-display/index.html (hunk 2 above)
Comment (sandersdan) [reply samples/video-decode-display/index.html:51]: Done.

File: samples/video-decode-display/index.html (hunk 3 above)
Comment (sandersdan) [reply samples/video-decode-display/index.html]: Removed entirely. I thought I had to use hasOwnProperty when iterating with `in`, but there was a separate bug that's now fixed (worker.js:10).

File: samples/video-decode-display/index.html (hunk 4 above)
Comment (sandersdan) [reply samples/video-decode-display/index.html:29]: Done.

**Comment by chcunningham** [comment]
@aboba @padenot I'll go ahead and merge since this just a sample. Feedback welcome.

**Pull request merged**